spec/models/lettings_log_spec.rb

```diff
@@ -1504,334 +1497,330 @@ def previous_postcode_fields
+        expect { lettings_log.update!(renewal: 1) }.to change(lettings_log, :offered).to 0
+        expect { lettings_log.update!(renewal: 0) }.to change(lettings_log, :offered).from(0).to nil
       end
 
-      it "correctly derives and saves prevten" do
-        lettings_log.managing_organisation.update!({ provider_type: "PRP" })
-        lettings_log.update!({ needstype: 1 })
-
-        record_from_db = ActiveRecord::Base.connection.execute("select prevten from lettings_logs where id=#{lettings_log.id}").to_a[0]
-        expect(record_from_db["prevten"]).to eq(32)
-        expect(lettings_log["prevten"]).to eq(32)
+      it "correctly derives referral if the letting is a renewal and clears it if it is not" do
+        expect { lettings_log.update!(renewal: 1) }.to change(lettings_log, :referral).to 1
+        expect { lettings_log.update!(renewal: 0) }.to change(lettings_log, :referral).from(1).to nil
+      end
 
-        lettings_log.managing_organisation.update!({ provider_type: "LA" })
-        lettings_log.update!({ needstype: 1 })
+      it "correctly derives voiddate if the letting is a renewal and clears it if it is not" do
+        expect { lettings_log.update!(renewal: 1) }.to change(lettings_log, :voiddate).to lettings_log.startdate
+        expect { lettings_log.update!(renewal: 0) }.to change(lettings_log, :voiddate).from(lettings_log.startdate).to nil
+      end
 
-        record_from_db = ActiveRecord::Base.connection.execute("select prevten from lettings_logs where id=#{lettings_log.id}").to_a[0]
-        expect(record_from_db["prevten"]).to eq(30)
-        expect(lettings_log["prevten"]).to eq(30)
+      it "correctly derives first_time_property_let_as_social_housing and clears it if it is not" do
+        expect { lettings_log.update!(renewal: 1) }.to change(lettings_log, :first_time_property_let_as_social_housing).to 0
+        expect { lettings_log.update!(renewal: 0) }.to change(lettings_log, :first_time_property_let_as_social_housing).from(0).to nil
       end
 
-      it "correctly derives and saves referral" do
-        record_from_db = ActiveRecord::Base.connection.execute("select referral from lettings_logs where id=#{lettings_log.id}").to_a[0]
-        expect(record_from_db["referral"]).to eq(1)
-        expect(lettings_log["referral"]).to eq(1)
+      it "correctly derives vacancy reason and clears it if it is not" do
+        expect { lettings_log.update!(renewal: 1) }.to change(lettings_log, :rsnvac).to 14
+        expect { lettings_log.update!(renewal: 0) }.to change(lettings_log, :rsnvac).from(14).to nil
       end
 
-      it "correctly derives and saves vacdays" do
-        record_from_db = ActiveRecord::Base.connection.execute("select vacdays from lettings_logs where id=#{lettings_log.id}").to_a[0]
-        expect(record_from_db["vacdays"]).to eq(0)
-        expect(lettings_log["vacdays"]).to eq(0)
+      # should have extra tests for whether it is derived when voiddate, mrcdate etc set and reset
```

Comment:
commented out line should be removed (as it looks like it's been addressed)

Reply:
true and true